eth/api.go

```diff
@@ -1197,6 +1217,13 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(txHash common.Hash) (ma
 		fields["contractAddress"] = receipt.ContractAddress
 	}
 
+	// Status field was introduced in ETH with EIP-609. For compatibility, we won't return status for
+	// blocks before the fork.
+	fields["status"] = nil
+	if receipt.Status != types.TxStatusUnknown && blockIndex >= 4370000 {
```

Comment:
If we keep this, I think it should be configurable via the chain config... somehow.

Since this implementation is only _in reference_ to a fork, but not actually a part of a fork, this is a strange case.

Otherwise, we just eliminate this condition and return status if available for all blocks regardless of chain or consensus schemes (since this deals only with the API, not protocol).

Reply:
I really don't like keeping this API method 100% compatible with ETH. 

IMHO we should keep one method (no extra methods in different namespaces), and "break" the compatibility (we're giving more functionality than ETH, and we're not compatible anyways right now).

Most useful solution will give us cleanest code.

Reply:
I agree, there is no goal to make it 100% compatible but to add useful feature for users